Duplicate cited_chunks when grounding=corpus

## Failure
When querying "what is adam claters job title and employer", the response contains 3 cited_chunks but only 1 unique chunk. The same chunk (133abba5-9eeb-5a99-8a5c-57e812c1ea83:2) appears 3 times in the cited_chunks array.

## Expected
cited_chunks should contain unique chunk identifiers without duplicates.

## Actual
cited_chunks: [
  {"doc_id": "133abba5-9eeb-5a99-8a5c-57e812c1ea83", "chunk_id": 2, "title": "Resume", ...},
  {"doc_id": "133abba5-9eeb-5a99-8a5c-57e812c1ea83", "chunk_id": 2, "title": "Resume", ...},
  {"doc_id": "133abba5-9eeb-5a99-8a5c-57e812c1ea83", "chunk_id": 2, "title": "Resume", ...}
]

## Relevant logs
2026-04-05 23:39:50,260 [ERROR] Invalid citation: query_hash=... doc_id=133abba5-9eeb-5a99-8a5c-57e812c1ea83 chunk_id=0 reason=not_in_retrieved_set

## Discovered in
E2E test run 20260405-1940

--- a/ragpipe/grounding.py
+++ b/ragpipe/grounding.py
@@ -357,7 +357,6 @@ def build_metadata(
     # reference the same chunk multiple times in a response.
     seen: set[tuple[str, int]] = set()
     cited_chunks = []
-    seen: set[tuple[str, int]] = set()
     for d, c in valid_citations:
         if (d, c) in seen:
             continue

